nixos/modules/services/web-apps/cryptpad.nix

```
@@ -0,0 +1,187 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.cryptpad;
+
+  inherit (lib) strings mdDoc mkOption types mkIf mkMerge;
+
+  # The Cryptpad configuration file isn't JSON, but a JavaScript source file that assigns a JSON value
+  # to a variable.
+  cryptpadConfigFile = builtins.toFile "cryptpad_config.js" ''
+    module.exports = ${builtins.toJSON cfg.settings}
+  '';
+
+  # Derive domain names for Nginx configuration from Cryptpad configuration
+  mainDomain = strings.removePrefix "https://" cfg.settings.httpUnsafeOrigin;
+  sandboxDomain = if cfg.settings.httpSafeOrigin == null then mainDomain
+    else strings.removePrefix "https://" cfg.settings.httpSafeOrigin;
+
+in {
+  options.services.cryptpad = {
+    enable = lib.mkEnableOption "cryptpad";
+
+    package = lib.mkPackageOption pkgs "cryptpad" {};
+
+    configureNginx = mkOption {
+      description = mdDoc ''
```

```suggestion
      description = ''
```

`lib.mdDoc` is deprecated.

---

Yeah just saw the manual build error, removed them.